networkit/cpp/io/NetworkitBinaryReader.cpp

```diff
@@ -45,6 +63,10 @@ Graph NetworkitBinaryReader::read(const std::string& path) {
         it += sizeof(uint64_t);
         memcpy(&header.offsetWeightTranspose, it, sizeof(uint64_t));
         it += sizeof(uint64_t);
+        memcpy(&header.offsetAdjIdLists, it, sizeof(uint64_t));
+        it += sizeof(uint64_t);
+        memcpy(&header.offsetAdjIdTranspose, it, sizeof(uint64_t));
+        it += sizeof(uint64_t);
```

Comment:
We need to make sure that these fields actually exist, do we? Old files will not have these entries.

Reply:
In fact, the header of old files will simply have the wrong size (since these fields are present whether edge IDs are in use or not). This probably requires a version bump.

Note that we still need to be able to read files of version 2 (and we should probably only write version 3 files if edge IDs are in use).

Reply:
I don't think it is necessary, since the size of the header shouldn't matter since we get a pointer(explicitly written) to where the data begins. It is the same case with unweighted graphs, where no weight data is written but we reader the header for it anyway, since it will be later handled with the header checker depending on the features flags.

yeah, so the size of the header doesn't matter because of the feature flags. since we will never read the pointers for IDs if that feature is not turned on, otherwise we just seek the data at the pointers specified in the header (not necessarily where the header immediately ends).

It is a similar situation to unweighted graphs where no weight data is written.

But I should test it.